File: go/friendly.go
package main

import "fmt"
import "os"
import "strconv"

func sumDevidors(victim uint64) (sum uint64) {
	sum = 1
	var i uint64
	for i=2; i*i < victim; i++ {
		if victim % i == 0 {
			sum += i + (victim / i)
		}
	}
	return
}

func main() {
	top, err := strconv.ParseUint(os.Args[1], 10, 64)
	if err != nil { panic(err) }
	var candidate, mate uint64
	for candidate = 1; candidate < top; candidate++ {
		mate = sumDevidors(candidate)
		if mate <= candidate { continue }
		if sumDevidors(mate) == candidate {
			fmt.Printf("%d,%d\n", candidate, mate)
		}
	}
}

File: rust/friendly.rs
use std::env;

fn sum_devidors(victim: i64) -> i64 {
  let mut sum: i64 = 1;
  let mut i: i64 = 2;
  while i*i <= victim {
    if victim % i == 0 {
      sum = sum + i;
      sum = sum + (victim / i);
    }
    i = i + 1;
  }
  return sum;
}

fn main() {
  let args: Vec<String> = env::args().collect();
  let top: i64 = args[1].parse::<i64>().unwrap();
  println!("top is {}", top);
  let mut mate: i64;
  for candidate in 1i64..top {
    mate = sum_devidors(candidate);
    if mate <= candidate {
      continue;
    }
    if sum_devidors(mate) == candidate {
      println!("{},{}", candidate, mate);
    }
  }
}
